### PR TITLE
Remove current-thread restriction on Orchestrator lock

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java [hunk 1]
```diff
@@ -108,55 +96,16 @@ public MutableStatusRegistry lockApplicationInstance_forCurrentThreadOnly(Applic
...
-        }
-    }
-
-    private void assertThreadDoesNotHoldLock(Thread currentThread, String message) {
-        synchronized (threadsHoldingLock) {
-            if (threadsHoldingLock.containsKey(currentThread)) {
-                throw new AssertionError(message + ", already have a lock on " + threadsHoldingLock.get(currentThread));
-            }
+            return new ZkMutableStatusRegistry(lock, applicationInstanceReference);
+        } catch (Throwable t) {
```
Comment (bjorncs) [inline orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: Use try-finally instead.

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java [hunk 2]
```diff
@@ -108,55 +96,16 @@ public MutableStatusRegistry lockApplicationInstance_forCurrentThreadOnly(Applic
     MutableStatusRegistry lockApplicationInstance_forCurrentThreadOnly(
             ApplicationInstanceReference applicationInstanceReference,
             long timeoutSeconds) {
-        Thread currentThread = Thread.currentThread();
-
-        //Due to limitations in SessionFailRetryLoop.
-        assertThreadDoesNotHoldLock(currentThread,"Can't lock " + applicationInstanceReference);
+        String lock2Path = applicationInstanceLock2Path(applicationInstanceReference);
```
Comment (bjorncs) [inline orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: Why is the lock still having the number "2" in its name? I thought the number's purpose was to distinguish from the old lock.

**Review by bjorncs — CHANGES_REQUESTED** [review]

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java (hunk 2 above)
Comment (hakonhall) [reply orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: Fixed

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java (hunk 1 above)
Comment (hakonhall) [reply orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: If an exception is NOT thrown I should not close it.

Comment (bjorncs) [reply orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: Ofc, my mistake 👍

**Review by bjorncs — APPROVED** [review]

**Pull request merged**